### PR TITLE
Bps 765 copy s2s secret to bulkscan vault

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-765

### Change description ###
Splitting #33 into 2 PRs as the build is failing to find the s2s secret in bulkscan key vault.
This PR copies the microservice secret from s2s vault to bulkscan vault.
Once this pr is merged, #33 build should be fixed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+  resource_group_name = "${local.s2s_rg}"
+}
+
+data "azurerm_key_vault_secret" "s2s_secret" {
+  key_vault_id = "${data.azurerm_key_vault.s2s_key_vault.id}"
+  name      = "microservicekey-bulk-scan-payment-processor"
+}
+
+# Copy s2s key from shared to local vault
+data "azurerm_key_vault" "local_key_vault" {
```
Comment (nitinprabhu) [inline infrastructure/main.tf]: may be just rename this to `bulk_scan_key_vault`

File: infrastructure/main.tf [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+  name                = "s2s-${var.env}"
+  resource_group_name = "${local.s2s_rg}"
+}
+
+data "azurerm_key_vault_secret" "s2s_secret" {
+  key_vault_id = "${data.azurerm_key_vault.s2s_key_vault.id}"
+  name      = "microservicekey-bulk-scan-payment-processor"
+}
+
+# Copy s2s key from shared to local vault
```
Comment (nitinprabhu) [inline infrastructure/main.tf]: copy s2s secret to bulk scan vault

**Comment by aliveni** [comment]
> Looks good to me but would be nice if we can fix review comments

Addressed review comments. Your review became stale, please approve again.

File: infrastructure/main.tf [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+}
+
+data "azurerm_key_vault" "s2s_key_vault" {
+  name                = "s2s-${var.env}"
+  resource_group_name = "${local.s2s_rg}"
+}
+
+data "azurerm_key_vault_secret" "s2s_secret" {
+  key_vault_id = "${data.azurerm_key_vault.s2s_key_vault.id}"
+  name      = "microservicekey-bulk-scan-payment-processor"
```
Comment (doncem) [inline infrastructure/main.tf]: mis-alignment. but not to worry

File: infrastructure/main.tf [hunk 4]
```diff
@@ -1,3 +1,32 @@
...
+}
+
+# Copy s2s secret from s2s key vault to bulkscan key vault
+data "azurerm_key_vault" "bulk_scan_key_vault" {
+  name = "${local.vaultName}"
```
Comment (doncem) [inline infrastructure/main.tf]: mis-alignment here too

Comment (doncem) [reply infrastructure/main.tf]: p.s. can avoid using "local" - just stick in vault name here directly

File: infrastructure/main.tf [hunk 5]
```diff
@@ -1,3 +1,32 @@
...
+  resource_group_name = "${local.vaultName}"
+}
+
+resource "azurerm_key_vault_secret" "bulk_scan_s2s_secret" {
+  name         = "microservicekey-bulk-scan-payment-processor"
```
Comment (doncem) [inline infrastructure/main.tf]: can we name as `s2s-secret-payment-processor` as discussed previously?

File: infrastructure/main.tf (hunk 4 above)
Comment (aliveni) [reply infrastructure/main.tf]: > just a name. but since that needs changing - might as well apply styling 🙂

renamed and applied style changes

**Pull request merged**